### PR TITLE
test: Add negative NaN test cases

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

File: test/unittests/execute_floating_point_test.cpp [hunk 1]
```diff
@@ -213,14 +222,33 @@ TYPED_TEST(execute_floating_point_types, binop_nan_propagation)
...
+
         EXPECT_THAT(execute(*instance, 0, {anan, cnan}), ArithmeticNaN(TypeParam{}));
+        EXPECT_THAT(execute(*instance, 0, {anan, -cnan}), ArithmeticNaN(TypeParam{}));
+        EXPECT_THAT(execute(*instance, 0, {-anan, cnan}), ArithmeticNaN(TypeParam{}));
+        EXPECT_THAT(execute(*instance, 0, {-anan, -cnan}), ArithmeticNaN(TypeParam{}));
+
         EXPECT_THAT(execute(*instance, 0, {cnan, anan}), ArithmeticNaN(TypeParam{}));
+        EXPECT_THAT(execute(*instance, 0, {cnan, -anan}), ArithmeticNaN(TypeParam{}));
+        EXPECT_THAT(execute(*instance, 0, {-cnan, anan}), ArithmeticNaN(TypeParam{}));
+        EXPECT_THAT(execute(*instance, 0, {-cnan, -anan}), ArithmeticNaN(TypeParam{}));
```
Comment (axic) [inline test/unittests/execute_floating_point_test.cpp:251]: Shouldn't we check the sign of the nan on the result side?

Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:251]: That is not requires as any sign is fine.

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp:251]: The spec restricts only the payload of the propagated NaN.

Comment (axic) [reply test/unittests/execute_floating_point_test.cpp:251]: > The spec restricts only the payload of the propagated NaN.

That means it could be any sign?

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp:251]: Yes.

Comment (axic) [reply test/unittests/execute_floating_point_test.cpp:251]: I mean in the above example, it is CPU specific whether `-cnan, -anan` will result in `anan` or `-anan`?

Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:251]: Both `cnan` and `-cnan` are wasm canonical NaNs. Similarly for arithmetic NaNs. The sign is ignored.

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp:251]: I don't know how CPUs propagate the sign bit, it probably depends on the opcode, too.

Comment (axic) [reply test/unittests/execute_floating_point_test.cpp:251]: I'd add a todo here to investigate this further.

Comment (chfast) [reply test/unittests/execute_floating_point_test.cpp:276]: I added a TODO comment.

**Pull request merged**